File: ios/Sources/CapacitorMunbynPlugin/CapacitorMunbynPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(CapacitorMunbynPlugin, "CapacitorMunbyn",
    CAP_PLUGIN_METHOD(connect, CAPPluginReturnPromise);
    CAP_PLUGIN_METHOD(disconnect, CAPPluginReturnPromise);
    CAP_PLUGIN_METHOD(newLine, CAPPluginReturnPromise);
    CAP_PLUGIN_METHOD(cut, CAPPluginReturnPromise);
    CAP_PLUGIN_METHOD(align, CAPPluginReturnPromise);
    CAP_PLUGIN_METHOD(fontSize, CAPPluginReturnPromise);
    CAP_PLUGIN_METHOD(print, CAPPluginReturnPromise);
    CAP_PLUGIN_METHOD(setLogo, CAPPluginReturnPromise);
    CAP_PLUGIN_METHOD(printLogo, CAPPluginReturnPromise);
    CAP_PLUGIN_METHOD(tableHeader, CAPPluginReturnPromise);
    CAP_PLUGIN_METHOD(checkConnection, CAPPluginReturnPromise);
)

File: android/src/main/java/hu/rahop/plugin/munbyn/CapacitorMunbynPlugin.java
package hu.rahop.plugin.munbyn;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "CapacitorMunbyn")
public class CapacitorMunbynPlugin extends Plugin {

    private CapacitorMunbyn implementation = new CapacitorMunbyn();

    @PluginMethod
    public void connect(PluginCall call) {
        call.unimplemented("Not implemented on Android.");
    }

    @PluginMethod
    public void disconnect(PluginCall call) {
        call.unimplemented("Not implemented on Android.");
    }

    @PluginMethod
    public void newLine(PluginCall call) {
        call.unimplemented("Not implemented on Android.");
    }

    @PluginMethod
    public void cut(PluginCall call) {
        call.unimplemented("Not implemented on Android.");
    }

    @PluginMethod
    public void print(PluginCall call) {
        call.unimplemented("Not implemented on Android.");
    }

    @PluginMethod
    public void addListener(PluginCall call) {
        call.unimplemented("Event listeners are not implemented on Android.");
    }

    @PluginMethod
    public void tableHeader(PluginCall call) {
        call.unimplemented("Not implemented on Android.");
    }

    @PluginMethod
    public void align(PluginCall call) {
        call.unimplemented("Not implemented on Android.");
    }

    @PluginMethod
    public void setLogo(PluginCall call) {
        call.unimplemented("Not implemented on Android.");
    }

    @PluginMethod
    public void printLogo(PluginCall call) {
        call.unimplemented("Not implemented on Android.");
    }

    @PluginMethod
    public void fontSize(PluginCall call) {
        call.unimplemented("Not implemented on Android.");
    }

    @PluginMethod
    public void checkConnection(PluginCall call) {
        call.unimplemented("Not implemented on Android.");
    }

    @PluginMethod
    public void toggleBold(PluginCall call) {
        call.unimplemented("Not implemented on Android.");
    }

    @PluginMethod
    public void toggleUnderline(PluginCall call) {
        call.unimplemented("Not implemented on Android.");
    }
}
